experimental/LieAlgebras/src/simple_lie_algebra.jl

```diff
@@ -0,0 +1,263 @@
+@attributes mutable struct SimpleLieAlgebra{C<:RingElement} <: LieAlgebra{C}
+#Construct a simple Lie algebra over a given ring with a given root system
+  mat_space::MatSpace
+  base_ring::Ring
+  root_system::RootSystem
+  dim::Int
+  s::Vector{Symbol}
+  root_type::Tuple{String,Int64}
+
+  function SimpleLieAlgebra{C}(R::Ring, S::String, cached::Bool) where {C<:RingElement}
+    RS = RootSystem(S)
+    di = number_of_roots(RS) + length(RS.simple_roots)
+    M = MatrixSpace(R, 1, di)
+    s = [Symbol("e_$i") for i in 1:di]
+    st = RS.root_system_type
+    return get_cached!(SimpleLieAlgebraDict, (R, RS, M), cached) do
+    new{C}(M, R, RS, di, s, st)
+    end::SimpleLieAlgebra{C}
+  end
+end
+
+const SimpleLieAlgebraDict = CacheDictType{
+  Tuple{Ring,RootSystem,MatSpace},SimpleLieAlgebra
+}()
+
+mutable struct SimpleLieAlgebraElem{C<:RingElement} <: LieAlgebraElem{C}
+  parent::SimpleLieAlgebra{C}
+  mat::MatElem{C}
+  function SimpleLieAlgebraElem{C}(
+    L::SimpleLieAlgebra{C},
+    M::MatElem{C}) where {C<:RingElement}
+    return new{C}(L, M)
+  end
+end
```

Comment:
```suggestion
mutable struct SimpleLieAlgebraElem{C<:RingElement} <: LieAlgebraElem{C}
  parent::SimpleLieAlgebra{C}
  mat::MatElem{C}
end
```
the default constructor needs not to be written down